src/api/auth/registerUser.ts

```diff
@@ -0,0 +1,67 @@
+import ses from "node-ses";
+import { v4 as uuidv4 } from "uuid";
+import bcrypt from "bcrypt";
+import { querySQL } from "./mysql";
+
+type InsertPromise = {
+  fieldCount: number;
+  affectedRows: number;
+  insertId: number;
+  serverStatus: number;
+  warningCount: number;
+  message: string;
+  protocol41: boolean;
+  changedRows: number;
+};
+
+const sesClient = ses.createClient({
+  key: process.env.AWS_SES_ACCESS_KEY_ID || "",
+  secret: process.env.AWS_SES_SECRET_ACCESS_KEY || "",
+  amazon: `https://email.eu-central-1.amazonaws.com`,
+});
+
+export const registerUser = async (
+  username: string,
+  password: string,
+  email: string
+) => {
+  const requestId = uuidv4();
+  const userId = uuidv4();
+
+  bcrypt.hash(password, 10).then(async function (result: string) {
+    const newUserSql = `INSERT INTO UsrAccount(Ua_id, Ua_login, Ua_email, Ua_password, VerificationId, EmailVerified) VALUES('${userId}', '${username}',
+    '${email}', '${result}', '${requestId}', 0)`;
+
+    const insertedUser = (await querySQL(newUserSql)) as InsertPromise;
+    console.log({ insertedUser });
```

Comment:
Since you appear to have changed the way users IDs are generated when registering them (from autoincremented field in the database, to using uid), this bit is no longer necessary, as it was used simply to retrieve information about last inserted user, to get his ID.

This would likely also make **InsertPromise type** obsolete as well.

Also, console.log() should be removed.